test/fixtures/depends-on.yaml

```diff
@@ -0,0 +1,107 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: depends.apps.flanksource.com
+spec:
+  group: apps.flanksource.com
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                replicas:
+                  type: integer
+                image:
+                  type: string
+  scope: Namespaced
+  names:
+    plural: depends
+    singular: depend
+    kind: Depend
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: samples.apps.flanksource.com
+spec:
+  group: apps.flanksource.com
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
```

Comment:
I think we need a status.conditions object in here, and then in the e2e test after verifying the deployment has not been created yet, update the conditions to be ready, and then wait for thre deployment to be created.